### ISSUE TITLE
is() doesn't handle dynamic type names

### ISSUE THREAD

**Issue by mccanne** [brimdata/zed#3247]
If a type name changes, `is()` should be able to match the different instances of the same name, e.g.,
```
% zq in.zson
{a:1}(=a)
{b:2}(=a)
{a:3}(=a)
% zq "is(type(a))" in.zson
{a:1}(=a)
{a:3}(=a)
```
zq should print all three records instead of just the one that matches the first type signature.

**Comment by philrz** [comment]
Still present in current Zed.

```
$ zq -version
Version: v1.2.0-20-g26dbda03

$ zq "is(<a>)" in.zson
{a:1}(=a)
{b:2}(=a)
```

**Comment by philrz** [comment]
Verified in Zed commit 520b145.

We can see this was fixed by the changes in #4078. Starting with Zed commit 520b145 that had the changes from that PR, all three records now appear in the output.

```
$ zq -version
Version: v1.2.0-48-g520b145a

$ zq "is(<a>)" in.zson
{a:1}(=a)
{b:2}(=a)
{a:3}(=a)
```

Thanks @nwt!
